2-mandatory/3-groceries.js

```diff
@@ -28,12 +27,20 @@ Exercise 1:
 // Gather all week item names into this array
 let weeklyGroceriesToBuy = [];
 
+let weeklyMealPlanValues = Object.values(weeklyMealPlan).flat();
+
+weeklyMealPlanValues.forEach((key) => {
+  if (!weeklyGroceriesToBuy.includes(key)) {
+    weeklyGroceriesToBuy.push(key);
+  }
+});
+
 /*
 Exercise 2:
   Loop through your list again, but now only collect the weekend items into the weekendGroceriesToBuy array.
 */
 // Gather weekend item names into this array
-let weekendGroceriesToBuy = [];
+let weekendGroceriesToBuy = Object.values(weeklyMealPlan.saturday).flat();
```

Comment:
You code passes test because 
1. There are no repeated item on Saturday 
2. there is no items on Sunday
So, it is necessary to get all items from Saturday and Sunday and go through the repeated item check again